snapshot/lib/snapshot/dependency_checker.rb

```diff
@@ -1,49 +1,54 @@
+require 'fastlane_core/device_manager'
+require 'fastlane_core/helper'
+
 module Snapshot
   class DependencyChecker
     def self.check_dependencies
-      return if Helper.test?
+      return if FastlaneCore::Helper.test?
 
       self.check_xcode_select
       self.check_simctl
     end
 
     def self.check_xcode_select
+      require_relative 'latest_os_version'
+
       unless `xcode-select -v`.include? "xcode-select version"
-        UI.error '#############################################################'
-        UI.error "# You have to install Xcode command line tools to use snapshot"
-        UI.error "# Install the latest version of Xcode from the AppStore"
-        UI.error "# Run xcode-select --install to install the developer tools"
-        UI.error '#############################################################'
-        UI.user_error!("Run 'xcode-select --install' and start snapshot again")
+        FastlaneCore::UI.error '#############################################################'
+        FastlaneCore::UI.error "# You have to install Xcode command line tools to use snapshot"
+        FastlaneCore::UI.error "# Install the latest version of Xcode from the AppStore"
+        FastlaneCore::UI.error "# Run xcode-select --install to install the developer tools"
+        FastlaneCore::UI.error '#############################################################'
+        FastlaneCore::UI.user_error!("Run 'xcode-select --install' and start snapshot again")
       end
 
       if Snapshot::LatestOsVersion.ios_version.to_f < 9 # to_f is bad, but should be good enough
-        UI.error '#############################################################'
-        UI.error "# Your xcode-select Xcode version is below 7.0"
-        UI.error "# To use snapshot 1.0 and above you need at least iOS 9"
-        UI.error "# Set the path to the Xcode version that supports UI Tests"
-        UI.error "# or downgrade to versions older than snapshot 1.0"
-        UI.error '#############################################################'
-        UI.user_error!("Run 'sudo xcode-select -s /Applications/Xcode-beta.app'")
+        FastlaneCore::UI.error '#############################################################'
```

Comment:
why isn't this kept as `UI` ? missing the `require module`?

Reply:
The problem here is that there is a circular dependency here, with the module requiring the dependency checker (which it uses inline) and the dependency checker requiring the module. That's also the reason why `latest_os_version` is only required in this method - because that in turn also requires the module.